sdk/src/beta9/sync.py

```diff
@@ -114,6 +117,160 @@ def _calculate_sha256(file_path: str, chunk_size: int = CHUNK_SIZE) -> str:
                 hasher.update(chunk)
         return hasher.hexdigest()
 
+    # The methods _match_file, _pattern_to_regex, and _translate_segment_glob are copied
```

Comment:
Maybe we should vendor this? This seems like overkill -- this isn't a big enough problem to warrant this much code.

Reply:
It makes it easier to keep up to date as well. Good with me